datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/MolecularRecord.kt

```diff
@@ -24,8 +26,9 @@ data class MolecularRecord(
     override val evidenceSource: String,
 ) : MolecularTest {
 
-    override fun testsGene(gene: String) =
-        if (experimentType == ExperimentType.HARTWIG_TARGETED) drivers.copyNumbers.any { gene == it.gene } else true
+    override fun testsGene(gene: String, molecularTestTargets: Predicate<List<MolecularTestTarget>>) =
+        if (experimentType == ExperimentType.HARTWIG_TARGETED) geneSpecifications!![gene]?.let { molecularTestTargets.test(it) }
+            ?: false else true
```

Comment:
I think it would be good to add tests for this. The `?: false else true` looks counterintuitive to me, though maybe it is correct, would be easier to tell with a test